.gitignore

```diff
@@ -22,10 +22,15 @@ cabal.project.local
 .ghc.environment.*
 .HTF/
 .hie/
+ghcid.txt
+
 # Stack
 .stack-work/
 stack.yaml.lock
 
+# Nix
+shell.nix
+
```

Comment:
I hope you don't mind me adding these. I like to use `nix-shell` to pull in all the project dependencies (such as ghc, and cabal) and use `ghcid` for development. I didn't want to clutter up anyone else's workflow so I added the files I use to .gitignore. If this is annoying I can just stage my files more carefully and remove this.